### PR TITLE
ENH: Add nice error message for unauthorized pushes

### LINKED ISSUE #246: authentication with dat push
0. ```export DAT_ADMIN_USER='hello' && export DAT_ADMIN_PASS='password'```
1. ```dat init && echo '{"key": 1, "vote_share": 51.33}' | dat import --json && dat listen```
2. clone it: ```cd .. && clone http://localhost:6461```
3. change some data: ```cd localhost && echo '{"key": 1, "vote_share" "NA"}' | dat import --json```
4. ```dat push```
```bash
Elapsed      : 0 s
Pushed       : 21 B      (0 B/s)
 - changes : 1
 - blobs   : 0

Push to remote has completed.
```
5. look at localhost:6461, no data has changed
6. ```dat push``` again, or even try ```dat push http://localhost:6461```
```bash
Elapsed      : 0 s
Pushed       : 21 B      (0 B/s)
 - changes : 1
 - blobs   : 0

Push to remote has completed.
```

If there is a username/password on the remote dat, we need to make this clearer when pushing to it and have the auth strategy available to the client

### PR DESCRIPTION
**Pull request by okdistribute**
fixes maxogden/dat#246

### CODE REVIEW

**Comment by karissa** [comment]
@mafintosh i want to prompt the user when this fail, should I do it here or watch for an error in dat?

File: package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
     "pump": "^0.3.5",
     "pumpify": "^1.3.2",
     "request": "^2.36.0",
-    "through2": "^0.6.1"
+    "through2": "^0.6.1",
+    "url": "^0.10.1"
```
Comment (mafintosh) [inline package.json]: url is a core module. we don't need to include it here.

Comment (karissa) [inline package.json]: done

**Comment by mafintosh** [comment]
awesome! the prompt should be in dat.

**Comment by mafintosh** [comment]
released in 3.3.0

**Pull request merged**